### ISSUE TITLE
Too many queries to test if hook is active

### ISSUE THREAD

**Issue by lmeyer1** [PrestaShop/PrestaShop#26747]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Prestashop is repeating this query several dozens of times
```
SELECT active
FROM `ps_hook` h
WHERE (h.name = "**") LIMIT 1
```
The screenshot are from Profiling mode on a product page:
![2021-11-25_111717](https://user-images.githubusercontent.com/7520695/143424749-7ccd8bfc-e566-47f7-8e4f-b5d45f790f53.jpg)
![2021-11-25_111754](https://user-images.githubusercontent.com/7520695/143424759-862b8d4a-7f45-42ed-be19-88cd7386ff71.jpg)
![2021-11-25_111816](https://user-images.githubusercontent.com/7520695/143424779-41b054ad-69b6-4f95-bb84-c48ae91b1a17.jpg)

### Expected behavior

Query once the table ps_hook. Then a simple `in_array($hook_name, $active_hooks)` will do it. This can save dozens of queries depending on the page, even more than hundred (/!\) on a category page.

Screenshot on the same product page with proposed PR:
The table ps_hook is queried 7 times instead of 74
![2021-11-25_113115](https://user-images.githubusercontent.com/7520695/143425335-7934ca5b-4422-4dec-80ac-7db421fa4b00.jpg)

### Steps to reproduce

1. Enable Profiling mode
2. Open a FO page, search on the page for PREFIX_hook, you'll see an impressive number
3. Scroll down to table stress: hook is the champion.

### PrestaShop version(s) where the bug happened

1.7.8.0

### PHP version(s) where the bug happened

7.3

**Comment by Hlavtox** [comment]
I love you

**Comment by hibatallahAouadni** [comment]
Hello @lmeyer1 

I manage to reproduce the issue only with PS1.7.8.0, 1.7.8.1 and develop, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/143432592-41349e94-0ac7-4ef6-9b81-97137d98c263.png)

The ps_hook is called 159 times (on develop, 276 times), while with older versions (like 1.7.7.8 and 1.7.6.7) it's called only 14 times:

![image](https://user-images.githubusercontent.com/23308427/143432723-d69a0bbe-3f0e-49e4-98a5-337d0c8cf1e7.png)

It's kind of a regression on 1.7.8.0! Thanks for your PR :rocket: 
Ping @PrestaShop/prestashop-core-developers wdyt?

Thank you

**Comment by Hlavtox** [comment]
@eternoendless I think we should find out who let this code enter the codebase. Prestashop will never be fast, if we are optimizing on one end and somebody adds 150 queries to every page load just like that. 👎

**Comment by alfonsolr09** [comment]
i cant reproduce it
https://www.awesomescreenshot.com/video/6170252?key=9152097cab8ac47f99a1b234e913460f

**Comment by lmeyer1** [comment]
@alfonsolr09 
Please read _Steps to reproduce_. You didn't enable _Profiling mode._
Read https://doc.prestashop.com/pages/viewpage.action?pageId=51184700 first.

**Comment by lmeyer1** [comment]
@hibatallahAouadni 
> It's kind of a regression on 1.7.8.0! 

This problem appeared in this version, because the possibility to deactivate hooks is new in this version, as Daniel stated in slack https://prestashop.slack.com/archives/CTKHSPWPM/p1637922790248200?thread_ts=1637773407.201500&cid=CTKHSPWPM

**Comment by Progi1984** [comment]
Fixed by @lmeyer1 in #26748 & validated by @hibatallahAouadni.
